File: C++/118. Pascal's Triangle.cpp
#include <vector>

using namespace std;

vector<vector<int>> generate(int numRows)
{
    vector<vector<int>> result;

    for (int i = 0; i < numRows; i++)
    {
        result.push_back(vector<int>(i + 1, 1));
        for (int j = 1; j < i; j++)
        {
            result[i][j] = result[i - 1][j - 1] + result[i - 1][j];
        }
    }
    return result;
}

/*

1
1 1
1 2 1
1 3 3 1
1 4 6 4 1

*/

File: Rust/118. Pascal's Triangle.rs
pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
    let mut pascal: Vec<Vec<i32>> = Vec::new();

    for i in 0..num_rows {
        pascal.push(Vec::new());
        for j in 0..i + 1 {
            if j == 0 || j == i {
                pascal[i as usize].push(1);
            } else {
                let value = pascal[(i - 1) as usize][(j - 1) as usize]
                    + pascal[(i - 1) as usize][j as usize];
                pascal[i as usize].push(value);
            }
        }
    }
    pascal
}
